### PR TITLE
SERVICE : Created IBasketService Interface and Basket Service Class

### PR DESCRIPTION
**Pull request by ssobirov3**

### CODE REVIEW

File: BelissimoCloneWPF.Service/DTOs/Baskets/BasketForViewDTO.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BelissimoCloneWPF.Service.DTOs.Baskets
+{
+    public class BasketForViewDTO
+    {
+
```
Comment (ShohruhUzDev) [inline BelissimoCloneWPF.Service/DTOs/Baskets/BasketForViewDTO.cs:12]: Remove this line

File: BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+using System.Linq.Expressions;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BelissimoCloneWPF.Service.Services.Baskets
+{
+    public class BasketService : IBasketService
+    {
+        private readonly IGenericRepository<Basket> _basketRepository;
```
Comment (ShohruhUzDev) [inline BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs:24]: change _baskerRepository to basketRepository

File: BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        private readonly IMapper mapper;
+
```
Comment (ShohruhUzDev) [inline BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs:26]: Remove this line

File: BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        public BasketService(IGenericRepository<Basket> basketRepository,IMapper mapper) 
+        {
+            this._basketRepository = basketRepository;
+            this.mapper = mapper;
+        }
+        public async ValueTask<BasketForViewDTO> CreateAsync(BasketForCreationDTO basketForCreationDTO)
```
Comment (ShohruhUzDev) [inline BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs:32]: put one line between constructor and method

File: BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        {
+            var Basket = await _basketRepository.CreateAsync(mapper.Map<Basket>(basketForCreationDTO));
```
Comment (ShohruhUzDev) [inline BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs:34]: change Basket to basket

File: BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+            await _basketRepository.SaveChangesAsync();
+
+            return mapper.Map<BasketForViewDTO>(Basket);
```
Comment (ShohruhUzDev) [inline BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs:37]: too change

File: BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+            var isDeleted = await _basketRepository.DeleteAsync(b => b.Id == id);
+            await _basketRepository.SaveChangesAsync();
+
+            if (!isDeleted)
+                throw new BelissimoCloneWPFException(404, "Basket not found");
+
+            return true;
+        }
+
+        public async ValueTask<IEnumerable<BasketForViewDTO>> GetAllAsync(PaginationParams @params, Expression<Func<Basket, bool>> expression = null)
```
Comment (ShohruhUzDev) [inline BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs:51]: must oshmasin one line 120 chararter

File: BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+            return mapper.Map<BasketForViewDTO>(basket);
+        }
+
+        public async ValueTask<BasketForViewDTO> UpdateAsync(int id, BasketForUpdateDTO basketForUpdateDTO)
+        {
+            var existingBasket = await _basketRepository.GetAsync(u => u.Id == id);
+
+            if (existingBasket == null)
+                throw new BelissimoCloneWPFException(404, "FoodOrder not found");
```
Comment (ShohruhUzDev) [inline BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs:73]: change FoodOrder to Basket

File: BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+            if (!isDeleted)
+                throw new BelissimoCloneWPFException(404, "Basket not found");
+
+            return true;
+        }
+
+        public async ValueTask<IEnumerable<BasketForViewDTO>> GetAllAsync(PaginationParams @params,
+            Expression<Func<Basket,
+            bool>> expression = null)
```
Comment (ShohruhUzDev) [inline BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs:47]: two line write one line

File: BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+
+namespace BelissimoCloneWPF.Service.Services.Baskets
+{
+    public class BasketService : IBasketService
+    {
+        private readonly IGenericRepository<Basket> basketRepository;
+        private readonly IMapper mapper;
+
```
Comment (ShohruhUzDev) [inline BelissimoCloneWPF.Service/Services/Baskets/BasketService.cs:19]: Remove this line

**Comment by MuhammadKasimov** [comment]
issue

**Pull request merged**